python/looker/rtl/user_session.py

```diff
@@ -61,48 +52,58 @@ def get_token(self):
             self.login()
         return self._token
 
-    def authenticate(self):
-         # TODO: still to be built based on request
-        return
+    def authenticate(self) -> Dict[str, str]:
+        token = self.get_token()
+        header = {}
+        if token and token.access_token:
+            header['Authorization'] = f'token {token.access_token}'
+        return header
 
     def login(self, user_id: Optional[str] = None) -> at.AuthToken:
         if not self.is_authenticated:
-            token = self.__login(user_id)
+            token = self._login(user_id)
             self._token = at.AuthToken(token)
-            return token
         return self._token
 
-    def __login(self, user_id: Optional[str]) -> tp.SDKSuccessResponse.value:
+    def _login(self, user_id: Optional[str]) -> tp.Response.value:
         if (user_id and self.user_id != user_id):
             # We are switching user ids
-            self.__logout()
+            self._logout()
 
-        self.__reset()
+        self._reset()
 
         # pylint: disable=too-many-function-args
-        result = self.__ok(self.transport.request(tp.HttpMethod.POST,
-                                                  '/login',
-                                                  body={
-                                                      'client_id': self.settings.client_id,
-                                                      'client_secret': self.settings.client_secret
-                                                  }))
+        result = self._ok(
+            self.transport.request(tp.HttpMethod.POST,
+                                   '/login',
+                                   body={
+                                       'client_id': self.settings.client_id,
+                                       'client_secret':
+                                       self.settings.client_secret
+                                   }))
         return result
 
     def logout(self) -> bool:
-        result = False
         if self.is_authenticated:
-            result = self.__logout()
-        return result
+            result = self._logout()
+        return bool(result)
 
-    def __logout(self):
+    def _logout(self) -> tp.Response.value:
         token = self._token
-        # TODO: this is still to be built based on authenticator
+        result = self._ok(
+            self.transport.request(
+                tp.HttpMethod.DELETE,
+                '/logout',
+                authenticator=(lambda: {
+                    'Authorization': f'token {token.access_token}'
+                }) if token else None))
+        return result
 
-    def __reset(self):
+    def _reset(self):
         self._token = at.AuthToken()
 
-    def __ok(self, response: Union[tp.SDKErrorResponse, tp.SDKSuccessResponse]
-            ) -> Union[tp.SDKErrorResponse, tp.SDKSuccessResponse.value]:
+    def _ok(self,
+            response: tp.Response) -> Union[tp.Response.value, se.SDKError]:
```

Comment:
I've been curious about how to annotate types of exceptions raised. I'm not sure there's support for that in annotations. I think this is saying "we'll return either a value or an error" but what you're actually doing is either returning a value or *raising* an error

Reply:
Good point, I agree. Do you think it should just say it returns a `tp.Response.value` then?

Reply:
yeah, I think the return type should just be `tp.ResponseValue` (see my new alias for that type). The function should still raise `se.SDKError` but I just don't think there's a way using typing/annotations to denote that.